src/blockdata/script.rs

```diff
@@ -287,6 +287,16 @@ impl Script {
     /// Creates a new empty script.
     pub fn new() -> Script { Script(vec![].into_boxed_slice()) }
 
+    /// Creates a script identical to `self` but with all instances of `OP_CODESEPARATOR` removed.
+    pub fn rm_op_codeseparator(&self) -> Script {
+        let v = self.0.iter()
+            .filter(|&op| op != &all::OP_CODESEPARATOR.into_u8())
+            .map(|&op| op)
+            .collect::<Vec<u8>>();
```

Comment:
I am very surprised that this passes the tests. This deletes every instance of the `CODESEPARATOR` byte, even if those bytes are not actually opcodes (e.g. they may be inside of datapushes).

Reply:
Correct me if I'm wrong but isn't that what this code does?
```
    /** Serialize the passed scriptCode, skipping OP_CODESEPARATORs */
    template<typename S>
    void SerializeScriptCode(S &s) const {
        CScript::const_iterator it = scriptCode.begin();
        CScript::const_iterator itBegin = it;
        opcodetype opcode;
        unsigned int nCodeSeparators = 0;
        while (scriptCode.GetOp(it, opcode)) {
            if (opcode == OP_CODESEPARATOR)
                nCodeSeparators++;
        }
        ::WriteCompactSize(s, scriptCode.size() - nCodeSeparators);
        it = itBegin;
        while (scriptCode.GetOp(it, opcode)) {
            if (opcode == OP_CODESEPARATOR) {
                s.write(AsBytes(Span{&itBegin[0], size_t(it - itBegin - 1)}));
                itBegin = it;
            }
        }
        if (itBegin != scriptCode.end())
            s.write(AsBytes(Span{&itBegin[0], size_t(it - itBegin)}));
    }
```

Reply:
ref: bitcoin/src/script/interpreter.cpp

Reply:
No, this code calls `scriptCode.GetOp` which parses the script.

Reply:
Oh I get you, `GetOp` (`GetScriptOp`) skips the push data but my code does not.